edgedb-client/src/credentials.rs

```diff
@@ -96,15 +121,26 @@ impl<'de> Deserialize<'de> for Credentials {
                 serde_json::to_string(&creds.tls_verify_hostname)
                     .map_err(de::Error::custom)?,
             )))
+        } else if creds.tls_ca.is_some() &&
+            creds.tls_cert_data.is_some() &&
+            creds.tls_ca != creds.tls_cert_data
+        {
+            Err(de::Error::custom(format!(
+                "detected conflicting settings: \
+                 tls_ca={} but tls_cert_data={}",
+                serde_json::to_string(&creds.tls_ca)
+                    .map_err(de::Error::custom)?,
+                serde_json::to_string(&creds.tls_cert_data)
+                    .map_err(de::Error::custom)?,
+            )))
```

Comment:
Using debug `{:?}` instead of serializing is fine here. Serializing is done for tls_security because options in tls_security have a bit different representation in Rust (i.e. CamelCase). While for strings it's mostly the same.